src/merkle/tiered_smt/index.rs

```diff
@@ -0,0 +1,59 @@
+use super::RpoDigest;
+
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub struct TreeIndex {
+    depth: u32,
+    index: u64,
+}
```

Comment:
I think a more correct name for this could be `NodeIndex`. Depth here should not exceed 64 and so we can use `u8` for it.

Overall, it might make sense to combine this with `BitsIterator`.

Reply:
`NodeIndex` is indeed better. Fixed!

As for the depth type, the only reason it is `u32` here is to maintain consistency with the other trees of the library (`SimpleSmt`, `MerkleTree`, etc). Given they will be an enum (`AdviceSet`), it is desirable to have an uniform type definition for them as the interfaces for this enum will be shared across these types.

`u8` sounds reasonable for all trees (simple/compact/merkle). Sounds good if I update to `u8` there as well?